IBPSA/Fluid/Actuators/Dampers/Exponential.mo

```diff
@@ -1,31 +1,54 @@
 within IBPSA.Fluid.Actuators.Dampers;
-model Exponential "Air damper with exponential opening characteristics"
-  extends IBPSA.Fluid.Actuators.BaseClasses.PartialDamperExponential(
-  final dp_nominal=(m_flow_nominal/kDam_default)^2,
-  dp(nominal=10),
-  final kFixed=0);
-protected
-   parameter Real kDam_default = sqrt(2*rho_default)*A/kThetaSqRt_default
-    "Flow coefficient for damper, k=m_flow/sqrt(dp), with unit=(kg*m)^(1/2)";
-   parameter Real kThetaSqRt_default(min=0)=
-     IBPSA.Fluid.Actuators.BaseClasses.exponentialDamper(
-       y=1,
-       a=a,
-       b=b,
-       cL=cL,
-       cU=cU,
-       yL=yL,
-       yU=yU)
-    "Flow coefficient, kThetaSqRt = sqrt(pressure drop divided by dynamic pressure)";
-initial algorithm
-   assert(kThetaSqRt_default>0, "Flow coefficient must be strictly positive.");
-   annotation (
+model Exponential
+  "Air damper with exponential opening characteristics"
+  extends IBPSA.Fluid.Actuators.BaseClasses.PartialDamperExponential;
+equation
+  // Pressure drop calculation
+  if linearized then
+    m_flow*m_flow_nominal_pos = k^2*dp;
+  else
+    if homotopyInitialization then
+      if from_dp then
+        m_flow=homotopy(
+            actual=IBPSA.Fluid.BaseClasses.FlowModels.basicFlowFunction_dp(
+                  dp=dp, k=k,
+                  m_flow_turbulent=m_flow_turbulent),
+            simplified=m_flow_nominal_pos*dp/dp_nominal_pos);
+      else
+        dp=homotopy(
+            actual=IBPSA.Fluid.BaseClasses.FlowModels.basicFlowFunction_m_flow(
+                  m_flow=m_flow, k=k,
+                  m_flow_turbulent=m_flow_turbulent),
+            simplified=dp_nominal_pos*m_flow/m_flow_nominal_pos);
+        end if;  // from_dp
+    else // do not use homotopy
+      if from_dp then
+        m_flow=IBPSA.Fluid.BaseClasses.FlowModels.basicFlowFunction_dp(
+                  dp=dp, k=k, m_flow_turbulent=m_flow_turbulent);
+      else
+        dp=IBPSA.Fluid.BaseClasses.FlowModels.basicFlowFunction_m_flow(
+                  m_flow=m_flow, k=k, m_flow_turbulent=m_flow_turbulent);
+      end if;  // from_dp
+    end if; // homotopyInitialization
+  end if; // linearized
+annotation (
 defaultComponentName="dam",
```

Comment:
I suggest to change to `defaultComponentName="damExp"`.  
Accordingly, I would add in the class `PressureIndependent` in the annotation section `defaultComponentName="damPreInd"`.  
This is not done in the valve models, but I think it would be a valuable feature for users.

Reply:
I agree. (And keep track of that for when I update the pressure independent valve model.)